### PR TITLE
Better support for CentOS/RHEL; Configurable Docker role; K8S 1.6.1

### PR DESCRIPTION
**Pull request by marccarre**
**Changelog**:
- Way of installing Docker can now be configured via optional Ansible variable.
- Improved support for CentOS/RHEL.
- Upgrade to Kubernetes 1.6.1 by default.
- Various bug fixes.

### CODE REVIEW

File: config_management/group_vars/all [hunk 1]
```diff
@@ -2,7 +2,8 @@
 go_version: 1.7.4
 terraform_version: 0.8.5
 docker_version: 1.11.2
-kubernetes_version: 1.5.2
+docker_install_role: 'docker-from-get.docker.com'
+kubernetes_version: 1.5.6
 kubernetes_cni_version: 0.3.0.1
```
Comment (brb) [inline config_management/group_vars/all]: AFAIK, k8s 1.5.6 requires k8s-cni 0.5.1

Comment (marccarre) [inline config_management/group_vars/all]: Correct, I get: `kubelet : Depends: kubernetes-cni (>= 0.5.1)`. Fixed now.

File: config_management/group_vars/all [hunk 2]
```diff
@@ -2,8 +2,9 @@
 go_version: 1.7.4
 terraform_version: 0.8.5
 docker_version: 1.11.2
-kubernetes_version: 1.5.2
-kubernetes_cni_version: 0.3.0.1
+docker_install_role: 'docker-from-get.docker.com'
+kubernetes_version: 1.5.6
```
Comment (brb) [inline config_management/group_vars/all]: Also, does k8s 1.5.6 work with the latest kubeadm?

File: config_management/group_vars/all (hunk 1 above)
Comment (marccarre) [inline config_management/group_vars/all]: Double checked and now installing fine:
```
[...]
TASK [kubernetes-install : install kubelet and kubectl] ************************
changed: [x.x.x.x] => (item=[u'kubelet=1.5.6*', u'kubectl=1.5.6*', u'kubernetes-cni=0.5.1*'])
changed: [y.y.y.y] => (item=[u'kubelet=1.5.6*', u'kubectl=1.5.6*', u'kubernetes-cni=0.5.1*'])
changed: [z.z.z.z] => (item=[u'kubelet=1.5.6*', u'kubectl=1.5.6*', u'kubernetes-cni=0.5.1*'])
[...]
```

File: config_management/group_vars/all (hunk 2 above)
Comment (marccarre) [inline config_management/group_vars/all]: Looks like it actually doesn't, well spotted.
Ansible was stuck, and when running manually I get an infinite loop of:
```
[apiclient] First node has registered, but is not ready yet
```
I'll move this branch back to K8S 1.5.2 as it seems to still be in the APT repository now, and CNI 0.3.0.1, until this gets resolved on `kubeadm`'s side.

**Comment by marccarre** [comment]
Kubernetes 1.6.1 + Weave Kube seems to work fine:
```
# kubectl --kubeconfig /etc/kubernetes/admin.conf get po --all-namespaces
NAMESPACE     NAME                             READY     STATUS    RESTARTS   AGE
kube-system   etcd-marc-0                      1/1       Running   0          6s
kube-system   kube-apiserver-marc-0            1/1       Running   0          6s
kube-system   kube-controller-manager-marc-0   1/1       Running   0          6s
kube-system   kube-dns-3913472980-9h4bg        3/3       Running   0          6m
kube-system   kube-proxy-5c9vf                 1/1       Running   0          6m
kube-system   kube-proxy-mvtjw                 1/1       Running   0          6m
kube-system   kube-proxy-t17gn                 1/1       Running   0          6m
kube-system   kube-scheduler-marc-0            1/1       Running   0          6s
kube-system   weave-net-fthlg                  2/2       Running   0          23s
kube-system   weave-net-rdmkd                  2/2       Running   0          23s
kube-system   weave-net-rk465                  2/2       Running   0          23s
```

**Comment by brb** [comment]
LGTM.

**Comment by brb** [comment]
@marccarre mind creating a PR to update the weave/tools submodule?

**Comment by marccarre** [comment]
Thanks for the review & merge @brb!
I'll create a PR to update `weave/tools` now.

**Pull request merged**